app/api_views.py

```diff
@@ -0,0 +1,71 @@
+from drf_spectacular.utils import extend_schema
+from rest_framework import status, viewsets
+from rest_framework.filters import SearchFilter
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .models import Currency, CurrencyPrice
+from .pagination import CurrencyPagination
+from .serializers import (
+    CurrencyConverterSerializer,
+    CurrencyDetailSerializer,
+    CurrencySerializer,
+)
+
+
+class CurrencyConverterView(APIView):
+    """
+    API view for currency conversion.
+    Methods:
+    - post: Handles POST requests and performs the currency conversion.
+    """
+
+    @extend_schema(request=CurrencyConverterSerializer, responses={200: 'converted_amount: float'})
+    def post(self, request, *args, **kwargs):
+        serializer = CurrencyConverterSerializer(data=request.data)
+        if serializer.is_valid():
+            from_currency = serializer.validated_data["from_currency"].upper()
+            to_currency = serializer.validated_data["to_currency"].upper()
+            amount = serializer.validated_data["amount"]
+
+            from_currency = CurrencyPrice.objects.filter(currency__symbol=from_currency).order_by("-created_at").first()
+            to_currency = CurrencyPrice.objects.filter(currency__symbol=to_currency).order_by("-created_at").first()
+
+            if from_currency is None or to_currency is None:
```

Comment:
you can use allow_null in serializer instead of checking it here. and if you want to check it here you should do it before hitting db